src/common/dsp/effect/ResonatorEffect.cpp

```diff
@@ -39,21 +46,152 @@ void ResonatorEffect::setvars(bool init)
+     * and the frequency of the particular band.
+     */
+    auto whichModel = *pdata_ival[resonator_mode];
+
+    int type = 0, subtype = 0;
+    switch (whichModel)
+    {
+    case 0:
+        type = fut_lp12;
+        subtype = 0;
+        // You need to adjust this switch statement
+        break;
+    default:
+        type = fut_none;
+        subtype = 0;
+        break;
+    }
+    FilterUnitQFPtr filtptr = GetQFPtrFilterUnit(type, subtype);
+
+    /*
+     * So now set up across the voices (e for 'entry' to match SurgeVoice) and the channels (c)
+     */
+    for (int e = 0; e < 3; ++e)
+    {
+        float frequencyInPitch = *f[resonator_freq1 + e * 3];
+        float resonance = *f[resonator_res1 + e * 3];
+        for (int c = 0; c < 2; ++c)
+        {
+            coeff[e][c].MakeCoeffs(frequencyInPitch, resonance, type, subtype, storage);
+
+            for (int i = 0; i < n_cm_coeffs; i++)
+            {
+                set1f(qfus[c].C[i], e, coeff[e][c].C[i]);
+                set1f(qfus[c].dC[i], e, coeff[e][c].dC[i]);
+            }
+
+            for (int i = 0; i < n_filter_registers; i++)
+            {
+                set1f(qfus[c].R[i], e, Reg[e][c][i]);
+            }
+
+            qfus[c].DB[e] = filterDelay[e][c];
+            qfus[c].WP[e] = WP[e][c];
+            qfus[c].WP[0] = subtype;
+
+            qfus[c].active[e] = 0xFFFFFFFF;
+            qfus[c].active[3] = 0;
+        }
+    }
+
+    /*
+     * FIXME - smooth this
+     */
+    float bankGain[3];
+    for (int i = 0; i < 3; ++i)
+    {
+        bankGain[i] = *f[resonator_gain1 + i * 3]; // probably some units wrk here
+    }
+
+    /* Run the filters. You need to grab a smoothed gain here rather than the hardcoded 0.3 */
+    for (int s = 0; s < BLOCK_SIZE_OS; ++s)
+    {
+        auto l128 = _mm_setzero_ps();
+        auto r128 = _mm_setzero_ps();
+
+        if (filtptr)
+        {
+            l128 = filtptr(&(qfus[0]), _mm_set1_ps(dataOS[0][s]));
+            r128 = filtptr(&(qfus[1]), _mm_set1_ps(dataOS[1][s]));
+        }
+        else
+        {
+            l128 = _mm_set1_ps(dataOS[0][s]);
+            r128 = _mm_set1_ps(dataOS[1][s]);
+        }
+
+        float tl[4], tr[4];
+        _mm_store_ps(tl, l128);
+        _mm_store_ps(tr, r128);
+        float mixl = 0, mixr = 0;
+        for (int i = 0; i < 3; ++i)
+        {
+            mixl += tl[i] * bankGain[i];
+            mixr += tr[i] * bankGain[i];
+        }
+
+        dataOS[0][s] = mixl;
+        dataOS[1][s] = mixr;
+    }
+
+    /* and preserve those registers and stuff. This all works */
+    for (int c = 0; c < 2; ++c)
+    {
+        for (int i = 0; i < n_filter_registers; i++)
+        {
+            for (int e = 0; e < 3; ++e)
+            {
+                Reg[e][c][i] = get1f(qfus[c].R[i], e);
+                coeff[e][c].C[i] = get1f(qfus[c].C[i], e);
+            }
+        }
+    }
```

Comment:
GCC doesn't seem to like this part. Just pasting this here, didn't look into it.
```
../src/common/dsp/effect/ResonatorEffect.cpp: In member function 'virtual void ResonatorEffect::process(float*, float*)':
../src/common/dsp/effect/ResonatorEffect.cpp:243:41: warning: iteration 8 invokes undefined behavior [-Waggressive-loop-optimizations]
  243 |                 coeff[e][c].C[i] = get1f(qfus[c].C[i], e);
      |                                    ~~~~~^~~~~~~~~~~~~~~~~
../src/common/dsp/effect/ResonatorEffect.cpp:238:27: note: within this loop
  238 |         for (int i = 0; i < n_filter_registers; i++)
      |                         ~~^~~~~~~~~~~~~~~~~~~~
```

Reply:
Oh man that’s a real bug. It should be r not c or n coefficient not registers. Will clean up today. Thanks. 

Which gcc version shows that? Surpsiede pipelines didn’t find

Reply:
GCC 10.2.1 20201203. The pipelines also show this, but we're not using `-Werror` on GCC.

Reply:
Anyway #3751 fixes both. Thank you!!!